src/constructRoutes.js

```diff
@@ -0,0 +1,48 @@
+import {
+  validateArray,
+  validateEnum,
+  validateKeys,
+  validateString,
+  validateObject,
+} from "./validation-helpers.js";
+
+export function constructRoutes(routesConfig) {
+  validate(routesConfig);
+
+  // TODO - construct a "resolved routes" return value
+}
+
+function validate(routesConfig) {
+  validateObject("routesConfig", routesConfig);
+
+  validateKeys("routesConfig", routesConfig, [
```

Comment:
Not sure if this is helpful or not for users of single-spa-layout. It prevents extra properties from being on the object passed into constructRoutes. Not sure if people who store this in db might be frustrated by it or not? Or if it would be helpful for those who put in extra properties without realizing that they won't work?

Reply:
We still allow custom props in the application. I think this is a good approach.

Reply:
After reviewing this some more I have a few more thoughts about this as I understood it more.

You're right this could cause problems if storing it in a DB and the DB puts extra values on it. It would force people to clean/sanitize it before sending it to use or get an error.

I wonder if instead of throwing an error we `console.warn` so people don't have to eliminate all extra properties but they are still informed that extra properties will be ignored.

Reply:
Console warn is probably better. And maybe even an option to turn off the warnings. I'll make the change